crates/core/tedge/src/cli/mqtt/publish.rs

```diff
@@ -32,7 +33,11 @@ impl Command for MqttPublishCommand {
 fn publish(cmd: &MqttPublishCommand) -> Result<(), MqttError> {
     let mut options = MqttOptions::new(cmd.client_id.as_str(), &cmd.host, cmd.port);
     options.set_clean_session(true);
-    let retain_flag = false;
+    let mut retain_flag = false;
+
+    if cmd.retain {
+        retain_flag = true;
+    }
```

Comment:
```suggestion
retain_flag = cmd.retain
```

You can even directly use `cmd.retain` to publish the message.

```
client.publish(&cmd.topic, cmd.qos, cmd.retain, payload)?;
```

Reply:
done.